### ISSUE TITLE
There is no "bounds" property on returned info (solved)

### ISSUE THREAD

**Issue by MocanuMarius** [lwansbrough/react-native-camera#683]
Hi, the bounds property is missing on android. Can we please get some answer to why that is so ? At least put a disclaimer there that it's not working.

Except that property, everthing else is in there.

### Environment
- **Node.js version**: 3.10.10
- **React Native version**: 0.42.3 (CLI 2.0.1)
- **React Native platform + platform version**: Android 5.1

### react-native-camera
**Version**: 0.7.0

**Comment by MocanuMarius** [comment]
After reviewing your android files i see that there is no real implementation of this using the Android Barcode api which should support this .. it's using some other api written by someone. Am i wrong here ?

**Comment by MocanuMarius** [comment]
Ok i have found the problem and the actual Zhing api or whatever .. i've put this piece of code in there
![image](https://cloud.githubusercontent.com/assets/6069369/25356535/1e6c189e-2943-11e7-8fce-01efeb537b8c.png)

Now i'm gonna see if i manage to rebuild this library of yours somehow... npm scripts are : [] :/

**Comment by MocanuMarius** [comment]
After spending the most usless hour in my programming life fiddling with a simple serialization of an Array in a stupid language, i have come to this solution that works: I will post my fork in a link below somewhere 
![image](https://cloud.githubusercontent.com/assets/6069369/25359886/9ab7d490-294f-11e7-9906-610bab058f47.png)

**Comment by MocanuMarius** [comment]
https://github.com/MocanuMarius/react-native-camera

**Comment by MocanuMarius** [comment]
After playing with it a little bit i realised that yes it works .. but the order those points are received at the x-y pairing they have is completely hard to understand ... :/ they are not corner points basically ,  they're just basically mingled togheter. Also keep in mind you have to use PixelRatio to get real screen space by dividing it from those points bounds.

**Comment by MocanuMarius** [comment]
.....
```
	readQR( e ) {
		let bounds = JSON.parse(e.bounds).map((bigArr) => bigArr.map(
			(item, index) => {
				// Some random width ratio that works for me
				if ( index == 0 ) return Math.floor(item/3)
				else return Math.floor(item/4)
				// Some random height ration that works for me
			}
		))
```

					<Image
						style={{
						opacity: this.state.opacity,
						height: this.state.bounds[0][1] - this.state.bounds[1][1],
						width: this.state.bounds[2][0] - this.state.bounds[1][0],
						top: this.state.bounds[1][1]*0.7,
						left: this.state.bounds[1][0] - 16,
						resizeMode: 'contain',
						position: 'absolute'
					}}
						source={{
						uri: this.state.parsedUrl
					}}/>

**Comment by MocanuMarius** [comment]
![screenshot_2017-04-25-04-19-07 1](https://cloud.githubusercontent.com/assets/6069369/25365036/f58b79d4-296e-11e7-9abf-bcebef946465.png)

**Comment by ptomasroos** [comment]
Will close since you consider it closed. And #631 already exists

**Comment by MocanuMarius** [comment]
I wouldn't close this... i wanted to leave it open to have a discussion on how to implement this in a PR.. i could make the PR for you but i need a review

**Comment by ptomasroos** [comment]
Sure thing.

**Comment by MocanuMarius** [comment]
@ptomasroos is it ok if we have a different bounds for android and iOS ? I would rather export the raw points given to me than calculating centers and origin and things like that you have for iOS. Does any of the maintainers have experience with Java ?

**Comment by gabox8888** [comment]
I feel like you are adding unnecessary serialization code. This should suffice 

event.putString("bounds", java.util.Arrays.toString(result.getResultPoints()));

**Comment by assafb81** [comment]
I did not understand how to add/set 'bound' property.
I want to do something like in the attached file - to focus the user on  a small box instead of entire screen
the result should look something like:

![qr-scanner-bounds](https://user-images.githubusercontent.com/28925534/28261692-f3b4fc00-6ae8-11e7-8c85-c262befc232a.png)

**Comment by MocanuMarius** [comment]
@assafb81 My print isn't focusing one region of the camera, it's detecting qr-codes everywhere. How you could do what you want though is by detecting codes everywhere, but only reacting when codes are within a certain bound in your camera view. How i would do this: 
1. Figure out the position and bounds of your "ok" box in screen-space.
2. Figure out the position on screen for your QR Code regardless of where it is ( like i did in this issue )
3. Only react if the qr is dected within the 1) bounds, using some maths.

**Comment by darioodiaz** [comment]
I was looking that issue, no bounds property on event, that solution is on a PR already? Amazing job boys!

**Comment by MocanuMarius** [comment]
@darioodiaz hey thanks..but since i'm not using this module anymore, (test project) ... someone else can pick up my code / PR ...someone with some Java knowledge prefferably and make a decent PR ..maybe try to sync with the other IOS interface. Until then...i guess what i provided above is a hack solution that works.

**Comment by darioodiaz** [comment]
@MocanuMarius until wednesday I'm busy but after that I'll try to do a PR @ptomasroos that feature are solved? or will be available in the next release? If not I'll check and try to do a PR

**Comment by blankg** [comment]
Would like to propose the following code:
```
                WritableArray resultPoints = Arguments.createArray();
                ResultPoint[] points = result.getResultPoints();
                if(points != null && points.length > 0) {
                    int pointsLength = points.length;
                    for(int index = 0; index < pointsLength; index++) {
                        ResultPoint point = points[index];
                        if(point != null) {
                            WritableMap newPoint = Arguments.createMap();
                            newPoint.putString("x", String.valueOf(point.getX()));
                            newPoint.putString("y", String.valueOf(point.getY()));
                            resultPoints.pushMap(newPoint);
                        }
                    }
                }
               event.putArray("bounds", resultPoints);

```
The result you will get in the javascript side would be an array of objects e.g. [{x: "222", y: "333"}, ...]

**Comment by blankg** [comment]
@darioodiaz, are you still planning to do a PR? if not let me know and I can do it.

**Comment by darioodiaz** [comment]
Hi everybody! I was and I'm too busy with my job and studies so I haven't time to do this, @blankg if you want to do the PR do it! I have much personal stuffs to do :( sorry boys

**Comment by ptomasroos** [comment]
Fixed in #845
